### PR TITLE
HELL OF A CHANGE 

### PR DESCRIPTION
**Pull request by fred15**
okay so i switched the mock transponder to feed in a List of lines from the files that match the same datestamp. had to create a datestamp variable so that i could actually match strings instead of the pingDatestamp which seem to get of the millisecs i think.

switched all the conflicting arguments in other classes/interfaces thus why its 10 files that i changed.

I created new tests for the mock transporter and did a simple fix on some of the other tests created by steven. 

Everything should be working now, I ran it, debuged it, ran the unit tests. I left a note in the OnPostDataEvent in dataprocessor for you steven I default all of that to the first item in  the list, you'll just need to create a for loop for it. 

anyhow whoever sees this please send me an email to confirm. I dont want another incident like Ben and mine happening again.

### CODE REVIEW

File: CollisionDetectionSystem/Interfaces/ITransponderReceiver.cs [hunk 1]
```diff
@@ -1,13 +1,14 @@
...
-	public delegate void DataDel(TransponderData data);
+	public delegate void DataDel( List<TransponderData> data);
 
 	public interface ITransponderReceiver
 	{
-		void ReceiveData(TransponderData data);
-		void PrepareDataForPost(TransponderData data);
+		void ReceiveData(List<TransponderData> data);
+		void PrepareDataForPost( List<TransponderData> data);
 		event DataDel PostDataEvent;
```
Comment (dsteeve) [inline CollisionDetectionSystem/Interfaces/ITransponderReceiver.cs:12]: Hey Fred, I don't understand why we want a List here.  This TransponderReceiver is suppose to receive just one transponderData at a time.  This in the real world would be the transponder getting one ping from a plane.  It would not get a list.  I will checkout out your branch and try to understand a bit more.

File: CollisionDetectionSystem/FunctionalObjects/MockTransponder.cs [hunk 2]
```diff
@@ -17,53 +17,86 @@ public class MockTransponder: IMockTransponder
...
+						//Console.WriteLine (line + "    " + count / 2); // prints line + the line number, ignores lines with #'s
 
 						string[] splitData = line.Split (',');
 
-						if (!(splitData.Length != 6))
-						{
-							
-							TransponderData tData = new TransponderData (splitData [0], splitData [1], double.Parse(splitData [2]), double.Parse(splitData [3]), double.Parse(splitData [4]), splitData [5]);
-							BroadcastDataEvent (tData);
-							System.Threading.Thread.Sleep(500);
```
Comment (dsteeve) [inline CollisionDetectionSystem/FunctionalObjects/MockTransponder.cs:57]: calling BroadcastDataEvent(tData) seems like what we wanted to be doing, I still don't get why you wanted to make this a list of transponderData elements?  That is not how the real transponder would work.  The mockTransponder is to mock what a real transponder would do.

  I realize you spent a lot of time to do this, but can you explain why you think this is the way it should work? What was wrong with the way it was working?

File: CollisionDetectionSystem/Interfaces/ITransponderReceiver.cs (hunk 1 above)
Comment (speppers) [inline CollisionDetectionSystem/Interfaces/ITransponderReceiver.cs:12]: If a transponder is tracking multiple planes doesn't it send data of all in range planes at a set interval at the same time? Other wise each data sent from the transponder would be off time wise by a couple of milliseconds per intruder so if there were 1000 intruders the last intruder to get sent would be significantly different time wise then the first?

File: CollisionDetectionSystem/FunctionalObjects/MockTransponder.cs (hunk 2 above)
Comment (fred15) [inline CollisionDetectionSystem/FunctionalObjects/MockTransponder.cs:57]: Eh only took about an hour and a half but it's what Steven wanted, basically
it's a lost of all events at a certain time stamp. So all events at a certain
time are sent in together instead of line by line.

I told him we'd agreed to only update when info on our aircraft came in but
yeah....

On Fri, Nov 20, 2015 at 10:13 AM, dsteeve < notifications@github.com [notifications@github.com] > wrote:
In CollisionDetectionSystem/FunctionalObjects/MockTransponder.cs
[https://github.com/dsteeve/teamDrei-4250/pull/19#discussion_r45492854] :

> ```
>                   string[] splitData = line.Split (',');
> ```
> -                     if (!(splitData.Length != 6))
> 
> ## \-                     {
> -                         TransponderData tData = new TransponderData (splitData [0], splitData [1], double.Parse(splitData [2]), double.Parse(splitData [3]), double.Parse(splitData [4]), splitData [5]);
> -                         BroadcastDataEvent (tData);
> -                         System.Threading.Thread.Sleep(500);

calling BroadcastDataEvent(tData) seems like what we wanted to be doing, I still
don't get why you wanted to make this a list of transponderData elements? That
is not how the real transponder would work. The mockTransponder is to mock what
a real transponder would do.

I realize you spent a lot of time to do this, but can you explain why you think
this is the way it should work? What was wrong with the way it was working?

—
Reply to this email directly or view it on GitHub
[https://github.com/dsteeve/teamDrei-4250/pull/19/files#r45492854] .[https://github.com/notifications/beacon/AIfWSsf6TqBYUpvKtmKHK5uLl-kl2Lvlks5pH0xNgaJpZM4GmM_5.gif]

File: CollisionDetectionSystem/Interfaces/ITransponderReceiver.cs (hunk 1 above)
Comment (fred15) [inline CollisionDetectionSystem/Interfaces/ITransponderReceiver.cs:12]: yeah i kinda agree, it seems more logical to send out all the info on airplanes who's info we received in the last second then send out every time you get one. I don't really know though.  

```
On Friday, November 20, 2015 10:23 AM, speppers <notifications@github.com> wrote:
```

 In CollisionDetectionSystem/Interfaces/ITransponderReceiver.cs:>

Comment (dsteeve) [inline CollisionDetectionSystem/Interfaces/ITransponderReceiver.cs:12]: A transponder is not what is tracking the planes.  https://en.wikipedia.org/wiki/Transponder_%28aeronautics%29
The transponder sits on an aircraft and sends out a ping when asked for a one.  We are mocking this situation (for now mocked for just our aircraft and one intruder to keep it simplistic).  

 Our transponderReceiver gets one ping from all the aircrafts around us because we are listening to their responses when asked by a radar sitting elsewhere.   What we are calling the DataProcessor was to keep a list of these data points by aircraft.

So, with that, I do see your point in that you want all the similar timestamps to trigger at the same time.  But, then we should NOT create a list but do a rapid fire of sending BroadCastDataEvent for the timestamps that are the same from both files to mimic real life.

Our data is pretty contrived, so to mimic that behavior might be as simple as to call BroadCastDataEvent for each file at the same time, and then doing the slight pause. 
So...if you are creating the lists by a minute's worth of timestamps, then in mockTransponder, call BroadCastDataEvent for each element in that list, but don't send a list to the transponderReceiver...would be my suggestion.

Comment (speppers) [inline CollisionDetectionSystem/Interfaces/ITransponderReceiver.cs:12]: List of a minutes worth of data? The list is just a convenient way to send 1 transponder data per plane in range. So if there were 2 intruders the list will have 3 elements, this aircraft data, intruder1 data, and intruder2 data. The data processor can then process all these at the same time and check for collisions otherwise each new intruder can throw the calculations off slightly since the math relies on how often the vectors are updated (when calculating time).

Comment (dsteeve) [inline CollisionDetectionSystem/Interfaces/ITransponderReceiver.cs:12]: okay, I see, not a minute's worth but 3 elements with all the same timestamps.  That is  good. However, TransponderReceiver.ReceiveData may have to put like things together  (if that is what you need), because in real life, the transponder from our aircraft, intruder1 aircraft and intruder2 aircraft will all send out pings from their transponders individually within possibly the "same second", and then the TransponderReceiver.receiveData(transponderData) would get all those individually.   I don't think we can change that part of the design, as that is a real life constraint.

What we do from there is a design choice, but we can't pretend that 3 different aircrafts are going to assemble them for us in a list for us.(they don't know about each other necessarily)

 I realize this list makes it easier for us to test, but we can't break the product in order for it to be easier to test.  

Am I making any sense? 
In other words, I see what you are saying, but TransponderReceiver.receiveData(transponderData) has to be left like that because it will only get one aircraft reading at a time.  
If TransponderReceiver wants to assemble things as a list or just call the PostDataEvent for each transponderData element is up to you.

I know that seems like a bunch of hoop jumping, but realize that MockTransponder is an object trying to mimic what the real world would be sending us.

Comment (dsteeve) [inline CollisionDetectionSystem/Interfaces/ITransponderReceiver.cs:12]: to be clear, let us go with these changes, I see what you were trying to do.   I will make a branch tonight to show what I am thinking, and let you review it tomorrow.  I don't want to derail your thoughts and I think the work Fred did to align the timestamps in a meaningful way makes sense.
